duckduckgo/Android
# Feature/only fire data clear pixel if manually clearing data

Task/Issue URL: https://app.asana.com/0/488551667048375/945922745759687
Tech Design URL: 
CC: 

**Description**:
Suppress fire pixel if the data is cleared automatically; also send for manual data clears

Also combines several test classes into one (which was unaddressed PR suggestion from #405)

**Steps to test this PR**:
1. Configure a MitM / Charles etc...
1. Use the 🔥 button from Browser activity; verify the `mf` pixel is fired
1. Use the 🔥 button from Tab switcher activity; verify the `mf` pixel is fired
1. Configure the automatic data clear options to clear everything after 5 seconds
1. Hit the home button, and wait 10s
1. Relaunch the app; verify the `mf` pixel **is not** sent for this


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
